### ISSUE TITLE
`reportInvalidCast` false positive when casting function to object

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#1176]
### Description

Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoDGAhgM4wCwAUFVQCYCmwUwYYAFAJQBcUAdPzUoB9KAF5CpGGzAAjAFb0CMADTNWHIA)

```python
from typing import cast


def foo(): ...

_ = cast(object, foo)
```
```
Conversion of type `() -> None` to type `object` may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to `object` first.
```

**Comment by DetachHead** [comment]
prob caused by https://github.com/microsoft/pyright/issues/10253
